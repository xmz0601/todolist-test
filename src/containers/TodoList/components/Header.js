import React, { Component } from 'react';

class Header extends Component {
  state = {
    value: ''
  };

  handleInputChange = (e) => {
    this.setState({
      value: e.target.value
    });
  };

  handleInputKeyUp = (e) => {
    if (e.keyCode === 13 && this.state.value.trim()) {
      this.props.addUndoItem(this.state.value.trim());
      this.setState({
        value: ''
      });  
    }
  };

  render() {
    return (
      <div className="header">
        <div className="header-content">
          <span className="header-text">TodoList</span>
          <input
            className="input-box"
            placeholder="Todo"
            data-test="input-box"
            value={this.state.value}
            onChange={this.handleInputChange}
            onKeyUp={this.handleInputKeyUp}
          />
        </div>
      </div>
    );
  }
}

export default Header;
