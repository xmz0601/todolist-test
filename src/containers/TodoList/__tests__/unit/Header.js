import React from 'react';
import Header from '../../components/Header.js';
import { shallow } from 'enzyme';
import { findTestWrapper } from '../../../../utils/testUtils.js';

test('Header snapshot', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});

test('Component Header should contain an input box', () => {
  const wrapper = shallow(<Header />);
  const inputElement = findTestWrapper(wrapper, 'input-box');
  expect(inputElement.length).toBe(1);
});

test('The original value of input box shoule be empty', () => {
  const wrapper = shallow(<Header />);
  const inputElement = findTestWrapper(wrapper, 'input-box');
  expect(inputElement.prop('value')).toEqual('');
});

test('When user inputs a new value, the value of input box changes too', () => {
  const wrapper = shallow(<Header />);
  const inputElement = findTestWrapper(wrapper, 'input-box');
  inputElement.simulate('change', {
    target: { value: 'Learn Jest' }
  });
  expect(wrapper.state('value')).toEqual('Learn Jest');
});

test('When user presses enter, if the value of input box is invalid, do nothing', () => {
  // When user presses enter, a function (addUndoItem) delivered by Component TodoList will be called, new data will be added to Component TodoList
  const fn = jest.fn();
  const wrapper = shallow(<Header addUndoItem={fn} />);
  const inputElement = findTestWrapper(wrapper, 'input-box');
  wrapper.setState({ value: '' });
  inputElement.simulate('keyUp', {
    keyCode: 13
  });
  expect(fn).not.toHaveBeenCalled();
});

test('When user presses enter, if the value of input box is valid, function should be called', () => {
  const fn = jest.fn();
  const wrapper = shallow(<Header addUndoItem={fn} />);
  const inputElement = findTestWrapper(wrapper, 'input-box');
  wrapper.setState({ value: 'Learn Jest' });
  inputElement.simulate('keyUp', {
    keyCode: 13
  });
  expect(fn).toHaveBeenCalled();
  expect(fn).toHaveBeenLastCalledWith('Learn Jest');
});

test('When user presses enter, if the value of input box is valid, clear input box after adding new data to undoList', () => {
  const fn = jest.fn();
  const wrapper = shallow(<Header addUndoItem={fn} />);
  const inputElement = findTestWrapper(wrapper, 'input-box');
  wrapper.setState({ value: 'Learn Jest' });
  inputElement.simulate('keyUp', {
    keyCode: 13
  });
  const newInputElement = findTestWrapper(wrapper, 'input-box');
  expect(newInputElement.prop('value')).toBe('');
});
