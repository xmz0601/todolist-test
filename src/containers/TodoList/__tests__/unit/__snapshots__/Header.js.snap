// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<div
  className="header"
>
  <div
    className="header-content"
  >
    <span
      className="header-text"
    >
      TodoList
    </span>
    <input
      className="input-box"
      data-test="input-box"
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="Todo"
      value=""
    />
  </div>
</div>
`;
